File: problemas/ad-hoc/3109/3109.kt
import java.lang.Math
import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val n: Int = sc.next().toInt()
    val mesas = mutableListOf<Int>()
    for (i in 0..n) {
        mesas.add(i)
    }

    val q: Int = sc.next().toInt()
    for (i in 1..q) {
        val e: Int = sc.next().toInt()
        val a: Int = sc.next().toInt()

        if (e == 1) {
            val b: Int = sc.next().toInt()

            val temp = mesas[a]
            mesas[a] = mesas[b]
            mesas[b] = temp
        } else {
            var resposta = 0
            var mesa = mesas[a]
            while(mesa != a) {
                resposta += 1
                mesa = mesas[mesa]
            }

            println(resposta)
        }
    }
}

File: problemas/ad-hoc/3109/3109.c
#include <stdio.h>

int main()
{
    int mesas[1000];
    int N, Q, E, A, B;

    scanf("%d\n", &N);
    for (int i = 1; i <= N; ++i)
    {
        mesas[i] = i;
    }

    scanf("%d\n", &Q);
    for (int i = 0; i < Q; ++i)
    {
        scanf("%d ", &E);

        if (E == 1)
        {
            scanf("%d %d\n", &A, &B);
            int temp = mesas[A];
            mesas[A] = mesas[B];
            mesas[B] = temp;
        }
        else if (E == 2)
        {
            scanf("%d\n", &A);

            int redirecionamentos = 0;
            int funcionario_atual = mesas[A];
            while (funcionario_atual != A)
            {
                ++redirecionamentos;
                funcionario_atual = mesas[funcionario_atual];
            }

            printf("%d\n", redirecionamentos);
        }
    }

    return 0;
}